Clarify arrow function brace requirements in documentation #34006 (#34033)
mdn/content: files/en-us/web/javascript/reference/functions/arrow_functions/index.md

() => a + b + 100;
```

> The braces can only be omitted if the function directly returns an expression. If the body contains statements, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.

Reviewer: ```suggestion
The braces can only be omitted if the function directly returns an expression. If the body contains anything other than a single `return` statement, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
```

> The braces can only be omitted if the function directly returns an expression. If the body contains anything other than a single `return` statement, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.

Reviewer: ```suggestion
The braces can only be omitted if the function directly returns an expression. If the body has statements, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
```

Returning object literals using the expression body syntax `(params) => { object: literal }` does not work as expected.

> ```js

Reviewer: Why are you changing these?

Author: Sorry my bad, It was added by copilot

Reviewer: Please revert all other lines of changes and only keep the one that says "contains statements". AI is probably not going to do you good here.

Author: sure give me a min
